Unlock radio mode and channel selection in WiFi station mode

salvatore-coppola opened the pull request:
Now radio mode is unlocked in WiFi station mode.
Now in station mode it is possible to select the channel to use from the bottom drop-down menu.
If the network search function is used, the suggested channel will be checked against the current device regdom.

salvatore-coppola committed:
Now radio mode is unlocked in WiFi station mode.
Now in station mode it is possible to select the channel to use from the bottom drop-down menu.
If the network search function is used, the suggested channel will be checked against the current device regdom.

marcellorinaldo commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:
```diff
@@ -315,6 +315,12 @@
     @UiField
     HelpButton countryCodeHelp;
+    @UiField
+    Modal regDomModal;
```
I would name this regDomErrorModal to make it clear that it pops out only on error

marcellorinaldo commented:
Tested successfully on RaspberryPi4.

pierantoniomerlino commented on kura/org.eclipse.kura.web2/src/main/java/org/eclipse/kura/web/client/ui/network/TabWirelessUi.java:
```diff
@@ -1444,7 +1435,7 @@ public void onSuccess(List<GwtWifiHotspotEntry> result) {
     }
 
-    private int getSelectedChannelIndex(int channelValue) {
+    private int geChannelIndexFromValue(int channelValue) {
```
Please fix typo.

salvatore-coppola committed:
Update TabWirelessUi.java

Fixed problem when channel 1 is choosen.
Fixed typo in get method.
Renamed regdom modal.

salvatore-coppola committed:
Fixed regdom modal name.

salvatore-coppola committed:
Improved netWifiChannelMissingError message.
